src/contribute/design-documents.md

```diff
@@ -0,0 +1,87 @@
+---
+layout: section
+title: 'Beam Design Documents'
+section_menu: section-menu/contribute.html
+permalink: /contribute/design-documents/
+---
+
+# Design Documents
+This is a collection of documents that may or may not be up to date.
+
+## Documents by category
+### Project Incubation (2016)
+- Technical Vision [[doc](https://docs.google.com/document/d/1UyAeugHxZmVlQ5cEWo_eOPgXNQA1oD-rGooWOSwAqh8/edit)], [[slides](https://docs.google.com/presentation/d/1E9seGPB_VXtY_KZP4HngDPTbsu5RVZFFaTlwEYa88Zw)]
+- Repository Structure [[doc](https://docs.google.com/document/d/1mTeZED33Famq25XedbKeDlGIJRvtzCXjSfwH9NKQYUE)]
+- Flink runner: Current status and development roadmap [[doc](https://docs.google.com/document/d/1QM_X70VvxWksAQ5C114MoAKb1d9Vzl2dLxEZM4WYogo)]
+- Spark Runner Technical Vision [[doc](https://docs.google.com/document/d/1y4qlQinjjrusGWlgq-mYmbxRW2z7-_X5Xax-GG0YsC0)]
+- PPMC deep dive [[slides](https://docs.google.com/presentation/d/1uTb7dx4-Y2OM_B0_3XF_whwAL2FlDTTuq2QzP9sJ4Mg)]
+
+### Beam Model
+- Checkpoints [[doc](https://s.apache.org/FIWQ)]
+- A New DoFn [[doc](https://s.apache.org/a-new-dofn)], [[slides](https://s.apache.org/presenting-a-new-dofn)]
+- Splittable DoFn (Obsoletes Source API) [[doc](http://s.apache.org/splittable-do-fn)]
+  - Reimplementing Beam API classes on top of Splittable DoFn on top of Source API [[doc](https://s.apache.org/sdf-via-source)]
+  - New TextIO features based on SDF [[doc](http://s.apache.org/textio-sdf)]
+  - Watch transform [[doc](http://s.apache.org/beam-watch-transform)]
+  - Technical contract for runtime rebalancing of work using Splittable DoFn [[doc](https://s.apache.org/beam-breaking-fusion)]
```

Comment:
I would suggest to move this one under Portability / Fn API, with text "Checkpointing and splitting of Beam bundles over the Fn API, with application to SDF"

Reply:
Done